build.gradle

```diff
@@ -164,75 +79,47 @@ def allSubProjectsConfig = {
         bndVersion = '2.4.1'
     }
 
-    gradle.taskGraph.whenReady { graph ->
-        if (graph.hasTask(publish)) {
-            if (!project.hasProperty('keyId') || !project.hasProperty('secretKeyRingFile') || !project.hasProperty('password')) {
-                throw new GradleException('You need to provide signing params in order to sign artifacts')
-            }
+    if (project.name == 'alfresco-dynamic-extensions-repo' || project.parent.name == 'alfresco-dynamic-extensions-repo') {
+        return
+    }
 
-            def id = project.hasProperty('keyId') ? project.keyId : ''
-            def file = project.hasProperty('secretKeyRingFile') ? project.secretKeyRingFile : ''
-            def password = project.hasProperty('password') ? project.password : ''
+    apply plugin: 'java'
+    apply plugin: 'jacoco'
 
-            ext."signing.keyId" = id
-            ext."signing.secretKeyRingFile" = file
-            ext."signing.password" = password
-        }
+    task sourceJar(type: Jar, dependsOn: classes) {
+        classifier = 'sources'
+        from sourceSets.main.allSource
     }
-}
 
-// Configuration specific for projects that are split up into Alfresco-version subprojects
-def subSubProjectsConfig = {
-    afterEvaluate {
-        sourceSets {
-            if (project.plugins.hasPlugin("groovy")) {
-                main.groovy.srcDirs += "${project.parent.projectDir}/shared/src/main/groovy"
-                test.groovy.srcDirs += "${project.parent.projectDir}/shared/src/test/groovy"
-            }
-            main.java.srcDirs += "${project.parent.projectDir}/shared/src/main/java"
-            test.java.srcDirs += "${project.parent.projectDir}/shared/src/test/java"
-            main.resources.srcDirs += "${project.parent.projectDir}/shared/src/main/resources"
-            test.resources.srcDirs += "${project.parent.projectDir}/shared/src/test/resources"
-
-            if (project.hasProperty("excludedSharedFiles")) {
-                def excludeFilter = {
-                    file ->
-                        if (file.file.path.contains("shared") && project.ext.excludedSharedFiles.contains(file.name)) {
-                            return true
-                        }
-                }
-                // We should use a different source file for this Alfresco version, the shared source will be excluded
-                main.java.exclude excludeFilter
-                javadoc.exclude excludeFilter
+    task javadocJar(type: Jar, dependsOn: javadoc) {
+        classifier = 'javadoc'
+        from javadoc.destinationDir
+    }
+
+    [compileJava, compileTestJava, javadoc]*.options*.encoding = 'UTF-8'
```

Comment:
We're setting the encoding to `UTF-8` twice ? :-)

This is the "old-way" - the `withTask(...)` below is the newer/slightly-better way ?